### PR TITLE
Document some configuration gotchas

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-code-manager/issues/792

### CODE REVIEW

File: .env.dev [hunk 1]
```diff
@@ -1,4 +1,5 @@
+# This loads for `yarn dev`, not the hosted dev site.
+# See https://github.com/mozilla/addons-code-manager#configuration
 REACT_APP_API_HOST=https://addons-dev.allizom.org
 # CDN URL
 PUBLIC_URL=https://addons-code-manager-dev-cdn.allizom.org
-REACT_APP_UNSAFE_ERROR_SIMULATION=true
```
Comment (bobsilverberg) [inline .env.dev]: Why was this removed? Does it not do anything in a `yarn dev` environment?

File: .env.dev [hunk 2]
```diff
@@ -1,4 +1,5 @@
+# This loads for `yarn dev`, not the hosted dev site.
```
Comment (bobsilverberg) [inline .env.dev]: Maybe changing this to ```This loads for `yarn dev` only``` (i.e., add the word "only") would make it even clearer?

File: .env.stage [hunk 3]
```diff
@@ -1,4 +1,5 @@
+# This loads for `yarn stage`, not the hosted stage site.
```
Comment (bobsilverberg) [inline .env.stage]: Maybe changing this to ```This loads for `yarn stage` only``` (i.e., add the word "only") would make it even clearer?

File: README.md [hunk 4]
```diff
@@ -65,6 +65,15 @@ yarn storybook
...
 When developing a new component, always add a story for it. If you were creating a component like `src/components/StarIcon/index.tsx` then you'd put its story in `stories/StarIcon.stories.tsx`. The storybook server will automatically load files having the `.stories.tsx` suffix in this directory.
 
+## Configuration
+
+You can configure the app by defining environment variables in `.env` files, the standard way to [configure Create React App](https://facebook.github.io/create-react-app/docs/adding-custom-environment-variables#adding-development-environment-variables-in-env).
+
+- How to add a new environment variable
+  - Add it to `.env` with the `REACT_APP_` prefix.
+- How to override an environment variable for a hosted site
+  - Define the variable in the [corresponding puppet config](https://github.com/mozilla-services/cloudops-deployment/tree/master/projects/addons-code-manager/puppet/yaml/type). To define a variable for `code.addons-dev.allizom.org`, for example, you'd update `amo.code_manager.dev.yaml`. Adding a variable to `.env.dev` **will do nothing** since that only affects the `yarn dev` command.
```
Comment (bobsilverberg) [inline README.md]: Should this be "[corresponding puppet config file]", as it is a file that needs to be edited, correct?

File: .env.dev (hunk 1 above)
Comment (kumar303) [reply .env.dev]: Correct. I put it in there mistakenly thinking I was setting a var for the dev site.

File: .env.dev (hunk 2 above)
Comment (kumar303) [reply .env.dev]: Sure
